internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -112,7 +112,20 @@ func (h *Host) SetConfidentialUVMOptions(enforcerType string, base64EncodedPolic
 
 	h.securityPolicyEnforcer = p
 	h.securityPolicyEnforcerSet = true
-	h.uvmReferenceInfo = base64UVMReference
+	h.uvmReferenceInfo = r.EncodedUVMReference
+
+	if r.PodStartupFragment != "" {
+		fragmentData, err := os.ReadFile(r.PodStartupFragment)
```

Comment:
is it not an error if we are told that a policy fragment should exist but it doesnt?